SwissDataScienceCenter/renku-ui
# Prevent changes to a project while the core-service is modifying it.

# Screenshots

<img width="1131" alt="image" src="https://user-images.githubusercontent.com/1196411/156827358-cb25b1d7-8d4f-4c72-bccf-57d2e8d096a7.png">

<img width="1131" alt="image" src="https://user-images.githubusercontent.com/1196411/156827401-fe6e0221-373e-4994-b4dd-0f73949f42ab.png">


<img width="1131" alt="image" src="https://user-images.githubusercontent.com/1196411/156827320-4c7b864d-69b6-44f9-89f7-ff74915a7d0f.png">


<img width="1131" alt="image" src="https://user-images.githubusercontent.com/1196411/156827269-bb0f5308-60c1-46f6-8aae-4e9098a44b51.png">

<img width="1116" alt="image" src="https://user-images.githubusercontent.com/1196411/157658321-981c7c3e-6e4b-460f-aba9-4ac8e726f3df.png">



# Testing

The easiest way to test is to run in telepresence and change the implementation of `CoreServiceProjectMixin` in `Project.state.js`.

```
 const CoreServiceProjectMixin = {
   callCount: 0,
   async fetchProjectLockStatus() {
     const client = this.client;
     const gitUrl = this.get("metadata.httpUrl");
@@ -50,8 +51,14 @@ const CoreServiceProjectMixin = {
       this.set("lockStatus.error", lockStatus.error.reason);
       return lockStatus.error.reason;
     }
     // this.set("lockStatus", lockStatus.data.result);
     // return lockStatus.data.result;
     // TESTING -- DELETE
     const locked = CoreServiceProjectMixin.callCount < 4;
     const result = { locked };
     this.set("lockStatus", result);
     CoreServiceProjectMixin.callCount = CoreServiceProjectMixin.callCount + 1;
     return result;
   },
 };
```

The `lockStatus` is updated when the user navigates to the Project>Settings or the Project>Dataset pages, so after doing that a few times, the lock should be resolved is using the above code.

To test in action:

1. Fork https://renku-ci-ui-1720.dev.renku.ch/projects/ralf.grubenmann/test-8
2. Update the fork to the latest version of Renku. You will see the UI blocked while the upgrade is happening.

Fix #1603

/deploy renku=000-acceptance-tests #persist
